config/webpack.config.js

```diff
@@ -0,0 +1,43 @@
+const webpack = require("webpack");
+const path = require("path");
+const BabiliPlugin = require("babili-webpack-plugin");
+const CleanWebpackPlugin = require("clean-webpack-plugin");
+
+module.exports = {
+  entry: path.resolve(__dirname, "../index.js"),
+  output: {
+    path: path.resolve(__dirname, "../build"),
+    filename: "mathsteps.min.js",
+    libraryTarget: "umd",
+    library: "mathsteps"
+  },
+  externals: {
+    mathjs: "mathjs"
+  },
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        exclude: [/node_modules/, /scripts/, /test/, /Config/],
+        use: ["babel-loader"]
+      }
+    ]
+  },
+  plugins: [
+    new webpack.LoaderOptionsPlugin({ // Options for loader that will run on files
+      minimize: true,
+      debug: false,
+      options: {
+        context: __dirname
+      }
+    }),
+    new webpack.optimize.ModuleConcatenationPlugin(), // Webpack 3 (scope hoisting)
+    new webpack.optimize.UglifyJsPlugin(), // Minifier
+    new webpack.DefinePlugin({ // Production ready build
+      "process.env": {
+        NODE_ENV: JSON.stringify("production")
+      }
+    }),
+    new CleanWebpackPlugin([path.resolve(__dirname, "../build")]) // Prebuild (kinda hook)
```

Comment:
Can you update the comments in this file to explain a bit better what things do? `kinda hook` confuses me

I'm not super familiar with webpack but will look some of this stuff up - but if you could go through this config and explain why you chose this config that'd be good too - thanks!

Reply:
Don't worry! I will comment the configuration and will make it easier for you also to make changes afterwards.